### PR TITLE
00 props state

### PR DESCRIPTION
**Pull request by Aprillion**
1st half of the workshop (live coding)

### CODE REVIEW

File: 00-props-state/README.md [hunk 1]
```diff
@@ -22,7 +22,18 @@ Excercises (follow live coding / create your own example / help people around wh
...
+
+## Useful tips
+
+* `<input value=...` is called a [Controlled Component](https://reactjs.org/docs/forms.html#controlled-components) (by React),<br>
+  `<input defaultValue=...` is [Uncontrolled](https://reactjs.org/docs/uncontrolled-components.html) (controlled by DOM / browser)
+* `<A>{(x) => <B prop={x} />}</A>` feature is called [Render Props](https://reactjs.org/docs/render-props.html) (a.k.a "children as a function")
+* `<>` is syntax sugar for [`<React.Fragment>`](https://reactjs.org/docs/fragments.html#short-syntax)
+* for reusable components, see also distinction between smart/dumb (a.k.a logical/presentational) components, e.g. https://youtu.be/Y7es1vcib14
+
+`TODO: update tips during live coding if needed`
```
Comment (Aprillion) [inline 00-props-state/README.md:49]: @janbarta and others - feel free to add more links or write down notes during the workshop, so we can create a PR afterwards with all the useful info that will be discussed :)

Comment (Aprillion) [reply 00-props-state/README.md:49]: note to self: https://git.io/fpU2u == https://github.com/msd-code-academy/react-workshop/tree/master/00-props-state#content

**Review by ra100 — APPROVED** [review]

File: 00-props-state/search-example/src/router.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+export const RouterContext = React.createContext()
+
+const getRoute = () => window.decodeURI(window.location.hash.substr(1))
+
+export const withRouter = (Component) => {
+  // dummy example, do NOT use in real life
```
Comment (janbarta) [inline 00-props-state/search-example/src/router.js:8]: Just out of curiosity, why? :)

**Review by janbarta — APPROVED** [review]

File: 00-props-state/search-example/src/router.js (hunk 2 above)
Comment (Aprillion) [reply 00-props-state/search-example/src/router.js:8]: no unit tests => probably bugs hiding

e.g. missing `componentWillUnmount` to clean up the event listener

**Pull request merged**